web-components/src/components/button-group/ButtonGroup.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) Cisco Systems, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import reset from "@/wc_scss/reset.scss";
+import { customElement, html, LitElement, property, PropertyValues, query } from "lit-element";
+import styles from "./scss/module.scss";
+import { FocusMixin, SlottedMixin } from "@/mixins";
+import { Key } from "@/constants";
+
+type CustomHTMLButtonElement = HTMLButtonElement & { selected: boolean };
+
+@customElement("md-button-group")
+export class ButtonGroup extends SlottedMixin(FocusMixin(LitElement)) {
+  @property({ type: Number, reflect: true }) active = 0;
+  @property({ type: Number, reflect: true }) tabIndex = 0;
+  @query("slot[name='button']") buttonSlot?: HTMLSlotElement;
+
+  private _disabled = false;
+  @property({ type: Boolean, reflect: true })
+  get disabled() {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    const oldValue = this._disabled;
+    this._disabled = value;
+    this.setAttribute("aria-disabled", `${value}`);
+    if (value) {
+      this.tabIndex = -1;
+    } else {
+      this.tabIndex = 0;
+    }
+    this.requestUpdate("disabled", oldValue);
+  }
+
+  get slotElement() {
+    return this.buttonSlot;
+  }
+
+  static get styles() {
+    return [reset, styles];
+  }
+
+  protected firstUpdated(changedProperties: PropertyValues) {
+    super.firstUpdated(changedProperties);
+
+    this.setAttribute("role", "group");
+  }
+
+  private setButtonTabIndex() {
+    this.slotted.forEach(button => ((button as CustomHTMLButtonElement).tabIndex = -1));
+  }
+
+  protected updated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+    if (changedProperties.has("slotted")) {
+      this.setButtonTabIndex();
+      this.setFirstActive();
+    }
+  }
+
+  get selectedBtnValue() {
+    return (this.slotted[this.active] as CustomHTMLButtonElement).value;
+  }
+
+  private notifySelectedChange() {
+    this.dispatchEvent(
+      new CustomEvent("button-group-change", {
+        detail: {
+          selected: this.selectedBtnValue
```

Comment:
I tried console logging `this.selectedBtnValue` and it seems like it is never defined, when I change the selection of the button group.

Reply:
Thanks @elenagiere! Fixed